sklearn/metrics/tests/test_classification.py

```diff
@@ -2624,62 +2624,37 @@ def test_log_loss():
     )
     loss = log_loss(y_true, y_pred)
     loss_true = -np.mean(bernoulli.logpmf(np.array(y_true) == "yes", y_pred[:, 1]))
-    assert_almost_equal(loss, loss_true)
+    assert_allclose(loss, loss_true)
 
     # multiclass case; adapted from http://bit.ly/RJJHWA
     y_true = [1, 0, 2]
     y_pred = [[0.2, 0.7, 0.1], [0.6, 0.2, 0.2], [0.6, 0.1, 0.3]]
     loss = log_loss(y_true, y_pred, normalize=True)
-    assert_almost_equal(loss, 0.6904911)
+    assert_allclose(loss, 0.6904911)
 
     # check that we got all the shapes and axes right
     # by doubling the length of y_true and y_pred
     y_true *= 2
     y_pred *= 2
     loss = log_loss(y_true, y_pred, normalize=False)
-    assert_almost_equal(loss, 0.6904911 * 6, decimal=6)
-
-    user_warning_msg = "y_pred values do not sum to one"
-    # check eps and handling of absolute zero and one probabilities
-    y_pred = np.asarray(y_pred) > 0.5
-    with pytest.warns(FutureWarning):
-        loss = log_loss(y_true, y_pred, normalize=True, eps=0.1)
-    with pytest.warns(UserWarning, match=user_warning_msg):
-        assert_almost_equal(loss, log_loss(y_true, np.clip(y_pred, 0.1, 0.9)))
-
-    # binary case: check correct boundary values for eps = 0
-    with pytest.warns(FutureWarning):
-        assert log_loss([0, 1], [0, 1], eps=0) == 0
-    with pytest.warns(FutureWarning):
-        assert log_loss([0, 1], [0, 0], eps=0) == np.inf
-    with pytest.warns(FutureWarning):
-        assert log_loss([0, 1], [1, 1], eps=0) == np.inf
-
-    # multiclass case: check correct boundary values for eps = 0
-    with pytest.warns(FutureWarning):
-        assert log_loss([0, 1, 2], [[1, 0, 0], [0, 1, 0], [0, 0, 1]], eps=0) == 0
-    with pytest.warns(FutureWarning):
-        assert (
-            log_loss([0, 1, 2], [[0, 0.5, 0.5], [0, 1, 0], [0, 0, 1]], eps=0) == np.inf
-        )
```

Comment:
@lorentzenchr I'm a bit confused. Does removing ``eps`` (deprecated in #25299) means that now ``eps`` is always 0 or ``eps`` is always computed based on the dtype ? The previous "auto" seems to indicate the latter but in that case testing edge cases is no longer possible.

Reply:
See https://github.com/scikit-learn/scikit-learn/issues/24515#issuecomment-1339829705.
My opinion is that eps=0 is the correct behavior (who are we to judge uncalibrated predicted probabilities!). The consensus was more in the direction of dtype dependent.

Reply:
At least, the clipping should not happen here to me. If y_true=0 and y_pred=0, the result should be exactly 0. `xlogy(0, 0) = 0` (no warning).

The question is do we want to return `inf` when y_true != 0 and y_pred = 0, or a finite value. If the former, we should clip with eps=0, else we should clip the result of `xlogy` as suggested in https://github.com/scikit-learn/scikit-learn/issues/24515#issuecomment-1336216473

I would go with returning `inf`, but I don't know if rely on it being finite (maybe if *SearchCV and co), and the warning message said that eps will be non-zero in 1.5, so maybe we should better keep it as is.

Reply:
Indeed, I find the following weird:

```python
>>> log_loss([0, 1, 2], [[1, 0, 0], [0, 1, 0], [0, 0, 1]])
2.2204460492503136e-16
```

Shall to conditionally clip to `eps` only when `one_hot_encode(y_true) > 0` and clip to 0 otherwise?

Reply:
Actually whatever the decision on this, there should be a test to cover the case where `log_loss` reaches its minimum (perfect predictions), both in binary and multiclass settings.

Reply:
I'd rather leave this discussion for a separate issue/PR to (try to) keep the focus of this PR on the deprecations clean-up.

I added a test for perfect predictions that only checks that the result is close to 0 for now.